src/features/storycreator/SlideEditor.tsx

```diff
@@ -0,0 +1,252 @@
+import { Card, CardContent, CardMedia, TextareaAutosize, Typography } from '@mui/material';
+import { useEffect } from 'react';
+import ReactGridLayout from 'react-grid-layout';
+
+import { useAppDispatch, useAppSelector } from '@/features/common/store';
+import styles from '@/features/storycreator/storycreator.module.css';
+import {
+  addContent,
+  editContent,
+  removeContent,
+  selectContentBySlide,
+} from '@/features/storycreator/storycreator.slice';
+import uiStyles from '@/features/ui/ui.module.css';
+import { selectWindows } from '@/features/ui/ui.slice';
+import Window from '@/features/ui/Window';
+
+interface Layout {
+  i: string;
+  x: number;
+  y: number;
+  w: number;
+  h: number;
+  static: boolean;
+  isDraggable: boolean;
+  content: string;
+}
+
+interface DropProps {
+  type: string;
+  static: boolean;
+  props: any;
+}
+
+const rowHeight = 30;
+const height = 400;
+
+export default function SlideEditor(props: any) {
+  const myWidth = props.width;
+  const targetRef = props.targetRef;
+  const slide = props.slide;
+  const imageRef = props.imageRef;
+  const takeScreenshot = props.takeScreenshot;
+
+  const content = useAppSelector((state) => {
+    return selectContentBySlide(state, slide);
+  });
+
+  useEffect(() => {
+    takeScreenshot();
+  }, [content]);
+
+  const dispatch = useAppDispatch();
+
+  const windows = useAppSelector(selectWindows);
+
+  const removeWindowHandler = (id: any) => {
+    dispatch(removeContent({ i: id, story: slide.story, slide: slide.i }));
+  };
+
+  const onDrop = (i_layout, i_layoutItem, event) => {
```

Comment:
when playing around with this, `layoutItem` was sometimes `undefined` - is that expected?